sdk-core/src/main/java/care/data4life/sdk/RecordService.kt

```diff
@@ -610,24 +612,24 @@ class RecordService(
 
         builder.setDataKey(dataKey)
 
-        if (record.encryptedAttachmentsKey != null) {
+        if (record.encryptedAttachmentsKey is EncryptedKey) {
             builder.setAttachmentKey(
                     cryptoService.symDecryptSymmetricKey(
                             commonKey,
-                            record.encryptedAttachmentsKey
+                            record.encryptedAttachmentsKey!!
                     ).blockingGet()
             )
         }
 
         return builder.build(
-                if (record.encryptedBody == null || record.encryptedBody.isEmpty()) {
-                    null// Fixme: This is a potential bug
-                } else {
+                if (record.encryptedBody is String && record.encryptedBody!!.isNotEmpty()) {
```

Comment:
to avoid using force !!

```
        val body = record.encryptedBody
        return builder.build(
                if (body is String && body.isNotEmpty()) {
                    fhirService.decryptResource<T>(
                            dataKey,
                            tags,
                            body
                    )
                } else {
                    null// Fixme: This is a potential bug
                }
        ) as DecryptedBaseRecord<T>
```

Reply:
done